spec/lib/jobs/ucs-service-profile-discovery-spec.js

```diff
@@ -0,0 +1,202 @@
+            ucs: "10.000.12.000",
+            verifySSL: false
+        },
+        service: "ucs-obm-service"
+    };
+
+    var node = {
+        id: 'abc',
+        type: 'compute',
+        name: 'ls-testServiceProfile',
+        identifiers: [],
+        relations: [
+            {
+                "relationType": "associatedTo",
+                "info": null,
+                "targets": [
+                    "sys/chassis-4/blade-1"
+                ]
+            }
+        ],
+        obm: [obm]
+    };
+    
+    before(function() { 
+        helper.setupInjector([
+            helper.require('/lib/jobs/base-job.js'),
+            helper.require('/lib/jobs/ucs-service-profile-discovery.js'),
+            helper.require('/lib/utils/job-utils/ucs-tool.js'),
+            helper.di.simpleWrapper(waterline,'Services.Waterline')
+        ]);
+        waterline.nodes = {
+            create: sandbox.stub().resolves(node),
+            needOne: sandbox.stub().resolves(node),
+            needOneById: sandbox.stub().resolves(node),
+            updateOne: sandbox.stub().resolves(node)
+        };
+        waterline.obms = {
+            upsertByNode: sandbox.stub().resolves(obm)
+        };
+        Error = helper.injector.get('Errors');
+    });
+    
+    afterEach(function() {
+        sandbox.restore();
+    });
+    
+    beforeEach(function() {
+        var Job = helper.injector.get('Job.Ucs.Service.Profile.Discovery');
+        ucsJob = new Job({
+            uri:'https://10.123.45.678:7080/sys',
+            username:'user',
+            password:'pass',
+            ucs:"10.000.12.000"
+        }, {}, taskId);
+        
+        sandbox.stub(ucsJob.ucs);
+        ucsTool = ucsJob.ucs;
+
+        rootData = {
+            body: {
+                ServiceProfile: {
+                    members:[
+                        {
+                            "assoc_state": "associated",
+                            "associatedServer": "sys/chassis-3/blade-3",
+                            "name": "ls-TTTTTTTTTT",
+                            "path": "org-root/ls-TTTTTTTTTT"
+                        },
+                        {
+                            "assoc_state": "associated",
+                            "associatedServer": "sys/rack-unit-3",
+                            "name": "ls-Profile3",
+                            "path": "org-root/ls-Profile3"
+                        }
+                    ],
+                    org: "root"
+                }
+            }
+        };
+        waterline.nodes.updateOne.reset();
+    });
+    
+    describe('usc service profile discovery', function() {
+        it('should successfully run the job', function() {
+            ucsTool.clientRequest.onCall(0).resolves(rootData);
+            ucsTool.clientRequest.onCall(1).resolves(rootData);
+            return ucsJob._run()
+                .then(function() {
+                     expect(waterline.nodes.updateOne).to.be.called.twice;
```

Comment:
There is no `called.twice` or `called.once`, you should use `calledTwice` and `calledOnce`.

The `twice` or `once` actually has no assertion effect, actually you could append any arbitrary words after `called`. For example:
`expect(xxxx).to.be.called.notexisting` or `expect(xxx).to.be.called.foobar` will always make the mocha happy. so the mocha success doesn't reflect such error.